site/components/pages/Resources/FeaturedArticles/ArticlesSlider.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { FC, useEffect, useState, useRef } from "react";
+import ArrowForwardIosIcon from "@mui/icons-material/ArrowForwardIos";
+import ArrowBackIosNewIcon from "@mui/icons-material/ArrowBackIosNew";
+import { Article } from "./Article";
+
+import { useElementWidth } from "hooks/useElementWidth";
+import { FeaturedPost } from "lib/queries";
+
+import * as styles from "./styles";
+
+type Props = {
+  articles: FeaturedPost[];
+};
+
+export const ArticlesSlider: FC<Props> = (props) => {
+  const articleRef = useRef(null);
+  const sliderRef = useRef<HTMLDivElement>(null);
+
+  const [articleWidth] = useElementWidth(articleRef);
+  const [sliderWidth] = useElementWidth(sliderRef);
+  const [, setScrollInterval] = useState<NodeJS.Timeout>();
```

Comment:
I think `useRef` is a better fit for this, plus then it doesn't cause re-renders (unless you want re-render?)

Reply:
Done